lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -430,16 +435,55 @@ def current(self) -> Node:
                 unit_number=self._charm.unit_id,
                 temperature=current_node.get("attributes", {}).get("temp"),
             )
+
         except OpenSearchHttpError:
-            # we try to get the most accurate description of the node
-            conf_on_disk = self.config.load("opensearch.yml")
+
+            # we try to get the most accurate description of the node from the static config
+            conf = self.config.load("opensearch.yml")
+
+            # also, if possible we rely on the Deployment Description (databag)
+            deployment_desc = self._charm.opensearch_peer_cm.deployment_desc()
+
+            # Application Priority: Deployment Description
+            # Reason: No reason to re-construct the App object
+            #  - it's available 99% of scenarios
+            #  - it's the same object as a re-constructed one (i.e. no dynamic changes on App)
+            if deployment_desc is None:
+                app = App(id=conf.get("node.attr.app_id"))
+            else:
+                app = deployment_desc.app
+
+            # Roles (Temperature) Priority: local config
+            # Reason:
+            #  - Deployment Description is holding "expected state" (that may not be applied)
+            #  - Static config holds the currently applied settings
+            try:
+                roles = conf["node.roles"]
+            except KeyError:
+                if deployment_desc and deployment_desc.start == StartMode.WITH_PROVIDED_ROLES:
+                    roles = deployment_desc.config.roles
+                else:
+                    roles = GeneratedRoles
+
+            if not roles:
```

Comment:
This will never be true, given the previous `else`

Reply:
Valid call, sry, I overlooked (moe details on https://github.com/canonical/opensearch-operator/pull/412/files#r1752624651). Closing this one as solution referred in the other thread applies here.